pkg/clients/dynatrace/settings_logmonitoring.go

```diff
@@ -0,0 +1,127 @@
+package dynatrace
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube/logmonitoring"
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/utils"
+)
+
+type IngestRuleMatchers struct {
+	Attribute string   `json:"attribute,omitempty"`
+	Operator  string   `json:"operator,omitempty"`
+	Values    []string `json:"values,omitempty"`
+}
+
+type logMonitoringValue struct {
+	ConfigItemTitle string               `json:"config-item-title,omitempty"`
+	Matchers        []IngestRuleMatchers `json:"matchers,omitempty"`
+	Enabled         bool                 `json:"enabled,omitempty"`
+	SendToStorage   bool                 `json:"send-to-storage,omitempty"`
+}
+
+type logMonitoringItem struct {
+	ObjectID           string             `json:"objectId,omitempty"`
+	LogMonitoringValue logMonitoringValue `json:"value,omitempty"`
+}
+
+type posLogMonSettingsBody struct {
+	SchemaId      string             `json:"schemaId"`
+	SchemaVersion string             `json:"schemaVersion"`
+	Scope         string             `json:"scope,omitempty"`
+	Value         logMonitoringValue `json:"value"`
+}
+
+const (
+	LogMonitoringSettingsSchemaId = "builtin:logmonitoring.log-storage-settings"
```

Comment:
does it have to be pulbic?

Reply:
no, you're right, changed it here: https://github.com/Dynatrace/dynatrace-operator/pull/4053/commits/6a09f0c3111f5ba96d35d0bd25daf61e2824a777